### PR TITLE
Eventing team code guidelines conclusion

### PR DESCRIPTION
**Pull request by nachtmaar**
**Description**

Changes proposed in this pull request:

- Add conclusion from [eventing team code guidelines voting session](https://github.com/kyma-project/community/pull/628/files).

**Result of the voting sessions**
- Assertion libraries: [stretchr/testify](https://github.com/stretchr/testify) (4/4) votes
- Mocking Libraries: [stretchr/testify](https://github.com/stretchr/testify#mock-package) with [vektra/mockery](https://github.com/vektra/mockery) (5/5) votes
- Structured Logging: [ueber-go/zap](https://github.com/uber-go/zap) (3/3) votes
- Regression testing:
  - `use regression in the test name to indicate whether a test is a regression test ?` (1/2) votes
  - `link a PR or issue to the test so that you can understand even two years later, what the problem was at the time ?` (5/5 votes)
- Table-driven tests naming:
  - input arguments prefix: `given` (4/4) votes
  - output arguments prefix: `want` (3/4) votes

**Related issue(s)**

#guide
https://github.com/kyma-project/kyma/issues/12820

### CODE REVIEW

File: concepts/eventing-code-guidelines.md [hunk 1]
```diff
@@ -75,11 +54,6 @@ For these reasons, `uber-go/zap` is the **preferred** structured logging library
 
 Furthermore, consider using `github.com/kyma-project/kyma/common/logging/logger`, which provides further **abstraction** over `uber-go/zap`.
 
-<!-- voting options:
-1. [ueber-go/zap](https://github.com/uber-go/zap) // proposed by guidelines
```
Comment (raypinto) [inline concepts/eventing-code-guidelines.md:79]: Do we need a line stating the logger we must use?

File: concepts/eventing-code-guidelines.md [hunk 2]
```diff
@@ -459,15 +433,26 @@ Concentrating on the main concern makes the individual tests shorter, easier to
...
--->
 - Express intent of code: Code should be self-explanatory and documented. In addition, tests help in understanding the intention of the code by providing test cases where you can easily see the (1) input to the code and (2) the expected outcome.
 - Drive development: Yes, tests can help drive development. They enable you to run code that is otherwise hard to run because it needs a complex setup. Imagine that in order to test a Kubernetes controller, you need to build a Docker image, push the image, and deploy the controller. This is a very time-consuming approach. You should always aim for a **short feedback cycle**.
 <!-- inspired by https://fossa.com/blog/golang-best-practices-testing-go/#Why -->
 
+
+### Regression Tests
+
+Whenever you close a bug, add a test which prevents the bug from getting re-introduced (*regression*).
+In order to understand at any time why the tests exists, you must **link** it to an **issue** to provide further context (if an issue exists).
```
Comment (raypinto) [inline concepts/eventing-code-guidelines.md:444]:
```suggestion
In order to understand at any time why the tests exist, you must **link** the test to an **issue** to provide further context (if an issue exists).
```

File: concepts/eventing-code-guidelines.md (hunk 1 above)
Comment (nachtmaar) [reply concepts/eventing-code-guidelines.md:79]: We do already have that, check [structured logging](https://github.com/kyma-project/community/blob/4ebb59628325f2acc626e02e94dc28a02853e837/concepts/eventing-code-guidelines.md#structured-logging) section :)

Comment (raypinto) [reply concepts/eventing-code-guidelines.md:79]: Ah I missed it! All good 👍

**Review by raypinto — DISMISSED** [review]

File: concepts/eventing-code-guidelines.md [hunk 3]
```diff
@@ -459,15 +433,26 @@ Concentrating on the main concern makes the individual tests shorter, easier to
...
-1. link a PR or issue to the test so that you can understand even two years later, what the problem was at the time ?
--->
 - Express intent of code: Code should be self-explanatory and documented. In addition, tests help in understanding the intention of the code by providing test cases where you can easily see the (1) input to the code and (2) the expected outcome.
 - Drive development: Yes, tests can help drive development. They enable you to run code that is otherwise hard to run because it needs a complex setup. Imagine that in order to test a Kubernetes controller, you need to build a Docker image, push the image, and deploy the controller. This is a very time-consuming approach. You should always aim for a **short feedback cycle**.
 <!-- inspired by https://fossa.com/blog/golang-best-practices-testing-go/#Why -->
 
+
+### Regression Tests
+
+Whenever you close a bug, add a test which prevents the bug from getting re-introduced (*regression*).
```
Comment (NHingerl) [inline concepts/eventing-code-guidelines.md:443]:
```suggestion
Whenever you close a bug, add a test that prevents the bug from getting re-introduced (*regression*).
```

File: concepts/eventing-code-guidelines.md [hunk 4]
```diff
@@ -459,15 +433,26 @@ Concentrating on the main concern makes the individual tests shorter, easier to
...
+In order to understand at any time why the tests exist, you must **link** the test to an **issue** to provide further context (if an issue exists).
```
Comment (NHingerl) [inline concepts/eventing-code-guidelines.md:444]:
```suggestion
Provide context to help understanding the purpose of the test. For example, if a test is based on an issue, you must **link** the test to the **issue**.
```

**Review by NHingerl — APPROVED** [review]

**Review by raypinto — APPROVED** [review]

**Pull request merged**